sunkit_pyvista/plotter.py

```diff
@@ -354,22 +354,35 @@ def plot_quadrangle(self, bottom_left, top_right=None, width: u.deg = None,
         self.plotter.add_mesh(quad_block, color=color, **kwargs)
         self._add_mesh_to_dict(block_name='quadrangles', mesh=quad_block)
 
-    def plot_field_lines(self, field_lines, **kwargs):
+    def plot_field_lines(self, field_lines, color_func=None, **kwargs):
         """
         Plot magnetic field lines from `pfsspy`.
 
         Parameters
         ----------
         field_lines : `pfsspy.fieldline.FieldLines`
             Field lines to be plotted.
+        color_func : function
+            Function to get the color for each field line.
         **kwargs :
             Keyword arguments are handed to `pyvista.Plotter.add_mesh`.
         """
+        if not color_func:
+            def color_func(field_line):
+                color = {0: 'black', -1: 'tab:blue', 1: 'tab:red'}.get(field_line.polarity)
+                return color
+
         field_line_meshes = pv.MultiBlock([])
         for field_line in field_lines:
             grid = self._coords_to_xyz(field_line.coords.ravel())
             field_line_mesh = pv.StructuredGrid(grid[:, 0], grid[:, 1], grid[:, 2])
-            color = {0: 'black', -1: 'tab:blue', 1: 'tab:red'}.get(field_line.polarity)
+
+            color = color_func(field_line)
+            if isinstance(color, tuple):
+                color = list(color)
+                if len(color) == 4:
+                    color.pop()
```

Comment:
This would be better as
```suggestion
                    opacity = color[3]
                    color = color[:3]
```